src/app/security/maturity-model/utils/useUrlHash.ts

```diff
@@ -0,0 +1,96 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+
+import { useRouter, usePathname } from 'next/navigation'
+
+import type { Route } from 'next'
+
+export const HASH_SIGN = '#'
+
+export type SectionHash = `${typeof HASH_SIGN}${string}`
+type NewHashEvent = HashChangeEvent | CustomEvent<SectionHash>
+
+export function useUrlHash() {
+  const pathname = usePathname()
+  const router = useRouter()
+
+  const [hash, setHash] = useState(getHashFromWindow)
+
+  function updateHash(sectionId: string) {
+    const sectionHash: SectionHash = `${HASH_SIGN}${sectionId}`
+    const pathnameWithNewSectionHash = pathname + sectionHash
```

Comment:
Maybe we can use template literals here so it's consistent with line 21?
```suggestion
    const pathnameWithNewSectionHash = `${pathname}${sectionHash}`
```
We can also move that after the `sectionHash === hash` condition as we don't need it before? A minor optimisation, but also works with having consts in one place.

Reply:
Love it!